tools/fastdds/shm/clean.py

```diff
@@ -0,0 +1,191 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sub-Command Clean implementation."""
+
+import os
+import platform
+import re
+from pathlib import Path
+
+if os.name == 'posix':
+    import fcntl
+elif os.name == 'nt':
+    import win32con
+    import win32file
+    import pywintypes
+    import msvcrt
+
+
+class Clean:
+    """This command searches and deletes zombie SHM ports / segments."""
+
+    def run(self):
+        """Execute the clean."""
+        self.__ports_in_use = 0
+        self.__segments_in_use = 0
+
+        zombie_segments = self.__clean_zombie_segments()
+        zombie_ports = self.__clean_zombie_ports()
+
+        print('shm.clean:')
+        print(self.__ports_in_use, 'ports in use')
+        print(self.__segments_in_use, 'segments in use')
+        # each port has 3 files
+        print(int(len(zombie_ports) / 3), 'zombie ports cleaned')
+        # each segment has 2 files
+        print(int(len(zombie_segments) / 2), 'zombie segments cleaned')
+
+    def __shm_dir(self):
+        """Return the shm directory.
+
+        Returns:
+        Path: The path to the platform specific the SHM directory
+
+        """
+        # Windows
+        if os.name == 'nt':
+            shm_path = Path('c:\\programdata\\eprosima\\'
+                            'fastrtps_interprocess\\').resolve()
+        elif os.name == 'posix':
+            # MAC
+            if platform.mac_ver()[0] != '':
+                shm_path = Path('/private/tmp/boost_interprocess/').resolve()
+            # Linux
+            else:
+                shm_path = Path('/dev/shm/').resolve()
+        else:
+            raise RuntimeError(f'{os.name} not supported')
+
+        return shm_path
+
+    def __list_dir(self):
+        """Return a list of files in the default SHM dir."""
+        try:
+            return os.listdir(self.__shm_dir())
+        except BaseException:
+            return []
+
+    def __clean_zombie_segments(self):
+        """Find & delete zombie segments in the default SHM dir.
+
+        Returns:
+        list(str): With the deleted file names
+
+        """
+        segment_lock_re = re.compile('^fastrtps_(\\d|[a-z]){16}_el|_sl')
+
+        # Each segment has an "_el" lock file that is locked if the segment
+        # is open and the owner process is alive
+        segment_locks = [
+            str(self.__shm_dir() / file_name) for file_name in self.__list_dir()
+            if segment_lock_re.match(file_name)]
```

Comment:
Just short enough for the linter
```suggestion
            str(self.__shm_dir() / filename) for filename in self.__list_dir()
            if segment_lock_re.match(filename)]
```